test/cppsuite/test_harness/workload_generator.h

```diff
@@ -149,20 +149,24 @@ class workload_generator : public component {
     static void
     execute_operation(thread_context &context)
     {
+        WT_SESSION *session;
         thread_operation operation;
 
-        // Set up work
-        operation = context.get_thread_operation();
+        session = connection::instance().get_session();
 
-        // Main loop
+        /* Main loop for worker threads. */
         while (context.is_running()) {
-            switch (operation) {
+            switch (context.get_thread_operation()) {
             case thread_operation::INSERT:
                 /* Sleep until it is implemented. */
                 std::this_thread::sleep_for(std::chrono::seconds(1));
                 break;
             case thread_operation::READ:
-                read_operation(context);
+                /*
+                 * This is a bit weird because read operation itself loops on the context running.
+                 * Will be improved later.
```

Comment:
You could remove the while loop from the read_operation function.
But opening too many cursors might cause an issue. They could be saved in the thread context and reused.

Reply:
Yeah we'll need to save the cursors in the thread context which is what I had thought about doing in the future, can do in this change.